packages/component-library-react/src/Calendar/index.tsx

```diff
@@ -48,78 +59,144 @@ function createCalendar(today: Date): Date[] {
-   *
-   * */
+   * `{date?: string; emphasis?: boolean; selected?: boolean; disabled?: boolean;}`
+   */
   events?: Events[];
   /**
-   * `currentDate` The default value is `new Date()`, but you can provide a different date
-   *   https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date
-   * */
-  currentDate?: Date;
+   * `showDate` The default value is current system date, but you can provide a different date as ISO 8601 date.
+   */
+  defaultValue?: string;
   /**
    * `locale` to change the calendar language by using 'date-fns/locale'
    *  `import { nl, enUS } from 'date-fns/locale';`
-   * */
+   */
   locale?: Locale;
   previousYearButtonTitle?: string;
   nextYearButtonTitle?: string;
   previousMonthButtonTitle?: string;
   nextMonthButtonTitle?: string;
+  min?: string;
+  max?: string;
+
+  /** @deprecated */
+  onCalendarClick?: (dateTime: string) => void;
+
+  /** @deprecated */
+  currentDate?: Date;
+
+  /** @deprecated */
   minDate?: Date;
+
+  /** @deprecated */
   maxDate?: Date;
 }
 
+const deprecated = (prev: string, current: string) => {
+  try {
+    console.warn(`${prev} is deprecated, please use: ${current}`);
+  } catch (e) {
+    // Ignore when no developers tools are available
+  }
+};
+
 /**
  * Calendar Component
  * powered by date-fns, so that make it easy to use the `date-fns` date functions & locale
  * */
 
 export const Calendar: FC<CalendarProps> = ({
-  onCalendarClick,
+  onChange,
   events,
-  currentDate,
+  defaultValue,
   locale = enUS,
   previousYearButtonTitle = 'Previous year',
   nextYearButtonTitle = 'Next year',
   previousMonthButtonTitle = 'Previous month',
   nextMonthButtonTitle = 'Next month',
+  onCalendarClick,
   minDate,
   maxDate,
+  min,
+  max,
+  currentDate,
 }) => {
-  const [visibleMonth, setVisibleMonth] = useState(currentDate || new Date());
-  const [selectedDate, setSelectedDate] = useState(currentDate);
+  let _minDate: Date, _maxDate: Date;
+
+  if (onCalendarClick) {
+    deprecated('onCalendarClick', 'onChange');
+  }
+  if (minDate instanceof Date) {
+    deprecated('minDate', 'min');
+    _minDate = startOfDay(minDate);
+  }
+  if (maxDate instanceof Date) {
+    deprecated('maxDate', 'max');
+    _maxDate = endOfDay(maxDate);
+  }
+
+  if (typeof min === 'string') {
+    _minDate = startOfDay(parseISO(min));
+  }
+  if (typeof max === 'string') {
+    _maxDate = endOfDay(parseISO(max));
+  }
+
+  const now = new Date();
+  const _events: InternalDayState[] = (events || []).map((day: Events) => ({
+    ...day,
+    date: parseISO(day.date),
+  }));
+
+  let defaultDate = typeof defaultValue === 'string' ? parseISO(defaultValue) : undefined;
+
+  if (currentDate instanceof Date) {
+    deprecated('currentDate', 'defaultValue');
+    defaultDate = currentDate;
+  }
+
+  const [visibleMonth, setVisibleMonth] = useState(defaultDate || now);
+  const [selectedDate, setSelectedDate] = useState(defaultDate);
```

Comment:
I think I also noticed an issue where when the calendar is visible and the `currentDate` prop changes, the state does not update accordingly. It sort of works for us because when you type the date we close the calendar/date picker, but you can resolve this with:

```tsx
if (selectedDate != defaultDate) {
  setSelectedDate(defaultDate);
}
```

Reference for the state update while rendering: https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes